lib/MaxMind/DB/Writer/Tree.pm

```diff
@@ -450,6 +514,43 @@ node that this record points to.
 
 For empty records, there are no additional arguments.
 
+=head2 $tree->freeze_tree($filename)
+
+Given a file name, this method freezes the tree to that file. Unlike the
+C<write_tree()> method, this method does write out a MaxMind DB file. Instead,
+it writes out something that can be quickly thawed via the C<<
+MaxMind::DB::Writer::Tree->new_from_frozen_tree >> constructor. This is useful if
+you want to pass the in-memory representation of the tree between processes.
+
+=head2 MaxMind::DB::Writer::Tree->new_from_frozen_tree()
+
+This method constructs a tree from a file containing a frozen tree.
+
+This method require the following two parameters:
+
+=over 4
+
+=item * filename
+
+The filename containing the frozen tree.
+
+=item * map_key_type_callback
+
+This is a subroutine reference that is called in order to determine how to
+store each value in a map (hash) data structure. See L<DATA TYPES> below for
+more details.
+
+This needs to be passed because subroutine references cannot be reliably
+serialized and restored between processes.
+
+=back
+
+=head2 Caveat for Freeze/Thaw
+
+The frozen tree is more or less the raw C data structures written to disk. As
+such, it is very much not portable, and your ability to thaw a tree on a
+machine not identical to the one on which it was written is not guaranteed.
```

Comment:
We should also mention that it may not work across different versions of this module.